src/System.IO.Abstractions.TestingHelpers/MockFileData.cs

```diff
@@ -188,9 +191,16 @@ public FileSecurity AccessControl
             {
                 // FileSecurity's constructor will throw PlatformNotSupportedException on non-Windows platform, so we initialize it in lazy way.
                 // This let's us use this class as long as we don't use AccessControl property.
-                return accessControl ?? (accessControl = new FileSecurity());
+                var value = Extensibility.RetrieveMetadata("AccessControl:FileSecurity");
+                if (value is not FileSecurity fileSecurity)
+                {
+                    fileSecurity = new FileSecurity();
+                    Extensibility.StoreMetadata("AccessControl:FileSecurity", fileSecurity);
+                }
+
+                return fileSecurity;
```

Comment:
Similar suggestion  to the other MockDirectoryData one. [ Feel free to ignore]
```suggestion
                var value = Extensibility.RetrieveMetadata("AccessControl:FileSecurity") as FileSecurity fileSecurity;
                if (value == null)
                {
                    fileSecurity = new FileSecurity();
                    Extensibility.StoreMetadata("AccessControl:FileSecurity", fileSecurity);
                }

                return fileSecurity;
```

Reply:
Will change it and also similar issues in
- DirectoryInfoAclExtensions
- FileInfoAclExtensions
- FileStreamAclExtensions